### ISSUE TITLE
Provide custom build script

### ISSUE THREAD

**Issue by christoph-buente** [snowplow/snowplow-javascript-tracker#450]
The full js tracker comes to a whopping 76k minified. Having this fat js deployed on a customers website can be a show stopper. Can you provide a build script to strip out optional features to minify the file size? Maybe also switching from grunt-yui-compressor to grunt-contrib-uglify might result in a better compression.

**Comment by alexanderdean** [comment]
This is a nice idea! It would be nice to start moving certain buckets of functionalities to opt-in/out plugins...

**Comment by fblundun** [comment]
I agree - for example, people who don't need to use form tracking could probably survive without the [forms.js file](https://github.com/snowplow/snowplow-javascript-tracker/blob/f7047658fbb6bc6ad235e1f6faeca8cc0bf34da0/src/js/forms.js).

On the choice of minifier: we originally chose to use grunt-yui-compressor because grunt-contrib-uglify has problems with certain non-printable characters ([source](https://github.com/mathiasbynens/grunt-yui-compressor#why-not-just-use-grunts-built-in-min-task-to-minify-javascript)). As far as I know this hasn't changed since.

**Comment by christoph-buente** [comment]
In general there are more options that might be optional. Like local storage and so on. I think the JS tracker is a special one. All the other backend tracker usually don't have a size or performance constraint. But here the tracker might sit on a webpage, that could suffer from long loading times and slow execution. I know it is all loaded asynchronously.

The point is, it lives on someone else's computer, not our own systems like any other tracker. How can we go from here? I think we don't have enough experience with the tracker to split the project up.

**Comment by rarkins** [comment]
+1

**Comment by alexanderdean** [comment]
Note: we would do #432, #541 first

**Comment by ryanrozich** [comment]
+1 - to add on to the benefits of this: Bundling would solve some performance/race conditions for us. 

We deploy snowplow script using the asynchronous loading pattern in our script, which our clients add to their sites. So on page load our script needs to load and then the snowplow script gets put into the browsers queue to download. We've seen strange data issues (that we've been working through on support tickets) that we think we can attribute back to waiting for the snowplow script to load so far later than our script loads. In some cases we think we are seeing users navigate away from the page before the snowplow script is loaded

If we could bundle the snowplow script with ours this would solve these issues because we'd know that snowplow is loaded at the same time as our script. It would be a big bonus if we could selectively decide which optional features to include because script size is a concern for our clients
